### DISCUSSION TITLE
Odin Notify brodcast error

### DISCUSSION THREAD

**Discussion by ihocan** [mbround18/valheim-docker#296] [Q&A] [answered]
## System Information

Operating System: Ubuntu 20.04 (LTS) x64
Docker Version: Docker version 20.10.5, build 55c4c88
Docker Compose Version: docker-compose version 1.27.4, build 40524192
Are you running with BepInEx? No

## Issue Description

I tried to send notification with Odin Notify but i'm faced below error, Start and Stop message working correctly. webhook url also correct.

> [ODIN][DEBUG] - Debugging set to true
> [ODIN][DEBUG] - Debug mode enabled!
> [ODIN][DEBUG] - Launching notify command...
> [ODIN][DEBUG] - Checking env for MESSAGE
> [ODIN][DEBUG] - Checking env for webhook_url
> [ODIN][DEBUG] - Env variable found webhook_url="webhookurl"
> [ODIN][INFO]  - Sending Broadcast: test
> [ODIN][DEBUG] - Webhook enabled, sending notification Broadcast
> [ODIN][DEBUG] - Webhook URL: "webhookurl"
> [ODIN][DEBUG] - Webhook Payload: {"event_type":{"name":"Broadcast","status":"Triggered"},"event_message":"test","timestamp":"2021-03-18T19:43:57.577771669+00:00"}
> [ODIN][ERROR] - [Broadcast]: Error with webhook! Status 500

**Comment by CosmicHorrorDev** [comment, marked as answer]
Unless you manually scrubbed the url in the logs, the webhook url has to be in the env var `WEBHOOK_URL`

**Reply by ihocan** [reply]
WEBHOOK_URL exist on enviroment and also it's correct one, i tested url with [Rust Playground](https://play.rust-lang.org/?version=stable&mode=debug&edition=2018&gist=42ac3d951b44767aba3d9b60f22d6bca), but i don't how it's happen when i use `odin notify "test"` command it's failing.

**Reply by CosmicHorrorDev** [reply]
How are you calling `odin_notify`? Are you in an interactive shell in the container, or are you just calling it with `docker exec`?

**Reply by ihocan** [reply]
I'm callin interactive shell with `docker exec -it valheimuser_valheim_1 bash` after that using `odin notify "test"` for broadcast

**Reply by CosmicHorrorDev** [reply]
And does the value for `echo $WEBHOOK_URL` look right? Somehow the url is getting messed up

**Reply by ihocan** [reply]
Yes, Url right and url in debug log also correct

**Reply by CosmicHorrorDev** [reply]
Are you on a recent version of the image? There was a problem with lingering double-quotes that was fixed about a week or so ago

**Reply by ihocan** [reply]
yes, i created server today, i think working latest one `sha256:06eb40108fb9aef3bb3465c1b667e6897988ed04fca8aa1c76e396fa4e8a931`

**Reply by CosmicHorrorDev** [reply]
Hmm, can you run

```bash
head -n 1 /home/steam/.version
```

from within the container? It's not picking up that it's a discord notification so as long as the url in the logs is valid (which you replaced with `webhook_url`). But if it's surrounded by double-quotes in the log then that would seem to be an old image

**Reply by ihocan** [reply]
when i run above command , it's return `f58c22c3e5e2b5ae692992005a89feb3da2963cf`
and my debug console like below, i jsut changed last words

> [ODIN][DEBUG] - Debugging set to true
> [ODIN][DEBUG] - Debug mode enabled!
> [ODIN][DEBUG] - Launching notify command...
> [ODIN][DEBUG] - Checking env for MESSAGE
> [ODIN][DEBUG] - Checking env for webhook_url
> [ODIN][DEBUG] - Env variable found webhook_url="https://discord.com/api/webhooks/822166652313272330/someid"
> [ODIN][INFO]  - Sending Broadcast: test
> [ODIN][DEBUG] - Webhook enabled, sending notification Broadcast
> [ODIN][DEBUG] - Webhook URL: "https://discord.com/api/webhooks/822166652313272330/someid"
> [ODIN][DEBUG] - Webhook Payload: {"event_type":{"name":"Broadcast","status":"Triggered"},"event_message":"test","timestamp":"2021-03-19T15:08:12.096357397+00:00"}
> [ODIN][ERROR] - [Broadcast]: Error with webhook! Status 500

and in envoriment parameter it's like below

> root@f73bd9302dae:/home/steam/steamcmd# echo $WEBHOOK_URL
> "https://discord.com/api/webhooks/822166652313272330/someid"

two url same as i see

**Reply by CosmicHorrorDev** [reply]
So if you're using a docker compose file you have a set of double-quotes around the webhook url. If you remove those then you shouldn't have any more issues.

Those should get stripped anyways and from all my testing they do so it seems like odin isn't fully up to date for you, or the quotes somehow aren't matching the quote character checked for `"`.

Regardless, removing the quotes around the webhook url in your docker compose file should make the issue go away.

**Reply by ihocan** [reply]
Okay when i remove double-quotes it's worked thank you for your help.
